### PR TITLE
fix(storybook__addon-storyshots): improve imageSnapshot options

### PR DESCRIPTION
**Pull request by bradleyayers**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by sheetalkamat** [comment]
@bradleyayers Can you please provide documentation for this change. Thank you.

**Comment by bradleyayers** [comment]
@sheetalkamat check out https://github.com/storybooks/storybook/tree/master/addons/storyshots#configure-storyshots-for-image-snapshots--alpha- as a good starting point. Beyond that I found jumping into the code useful https://github.com/storybooks/storybook/blob/master/addons/storyshots/src/test-body-image-snapshot.js

File: types/storybook__addon-storyshots/index.d.ts [hunk 1]
```diff
@@ -34,9 +34,11 @@ export interface ImageSnapshotOptions {
 }
 
 export function imageSnapshot(options?: {
-    storybookUrl: string;
-    getMatchOptions: (options: ImageSnapshotOptions) => { failureThreshold: number, failureThresholdType: 'percent' };
-    beforeScreenshot: (page: Page, options: ImageSnapshotOptions) => Promise<void>;
+    storybookUrl?: string;
+    getMatchOptions?: (options: ImageSnapshotOptions) => { failureThreshold: number, failureThresholdType: 'percent' };
+    getScreenshotOptions?: () => ScreenshotOptions;
```
Comment (sheetalkamat) [inline types/storybook__addon-storyshots/index.d.ts]: this needs to be
```typescript
getScreenshotOptions?: (options: ImageSnapshotOptions) => ScreenshotOptions
```

File: types/storybook__addon-storyshots/index.d.ts [hunk 2]
```diff
@@ -34,9 +34,11 @@ export interface ImageSnapshotOptions {
...
+    beforeScreenshot?: (page: Page, options: ImageSnapshotOptions) => Promise<void>;
+    getGotoOptions?: () => NavigationOptions;
```
Comment (sheetalkamat) [inline types/storybook__addon-storyshots/index.d.ts]: this needs to be 
```typescript
getGotoOptions?: (options: ImageSnapshotOptions) => NavigationOptions;
```

File: types/storybook__addon-storyshots/index.d.ts [hunk 3]
```diff
@@ -34,9 +34,11 @@ export interface ImageSnapshotOptions {
...
 }): Test;
```
Comment (sheetalkamat) [inline types/storybook__addon-storyshots/index.d.ts]: also need 
```typescript
chromeExecutablePath?: string;
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

**Comment by bradleyayers** [comment]
Thanks @sheetalkamat!

**Review by sheetalkamat — APPROVED** [review]

**Pull request merged**